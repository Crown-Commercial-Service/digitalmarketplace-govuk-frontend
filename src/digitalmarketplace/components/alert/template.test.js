const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('alert')

describe('alert', () => {
  it('default example passes accessibility tests', async () => {
    const $ = render('alert', examples.default)
    const results = await axe($.html())
    expect(results).toHaveNoViolations()
  })

  it('renders an alert component with the title text', () => {
    const $ = render('alert', examples.default)
    expect($.html()).toMatchSnapshot()
  })

  it('renders an alert component with the optional headingLevel param of 1', () => {
    const $ = render('alert', examples['with optional headingLevel param'])
    expect($.html()).toMatchSnapshot()
  })

  it('renders an alert component with error message', () => {
    const $ = render('alert', examples['with an error message'])
    expect($.html()).toMatchSnapshot()
  })
})
