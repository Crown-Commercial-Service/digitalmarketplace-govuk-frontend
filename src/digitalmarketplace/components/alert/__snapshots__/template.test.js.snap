// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert renders an alert component with error message 1`] = `
"<html><head></head><body><div data-module=\\"dm-alert\\" class=\\"dm-alert dm-alert--error dm-alert-test-class\\" dm-alert-test-key-1=\\"dm-alert-test-value-1\\" dm-alert-test-key-2=\\"dm-alert-test-value-2\\">
    <h2 class=\\"dm-alert__title\\">
        An action has been <span style=\\"color: red\\">completed</span> that lead to an error alert.
    </h2>
        <div class=\\"dm-alert__body\\">
            This is the <strong>HTML</strong> below the title.
        </div>
</div>
</body></html>"
`;

exports[`alert renders an alert component with the optional headingLevel param of 1 1`] = `
"<html><head></head><body><div data-module=\\"dm-alert\\" class=\\"dm-alert dm-alert--notice\\">
    <h1 class=\\"dm-alert__title\\">
        A successful action has been completed.
    </h1>
</div>
</body></html>"
`;

exports[`alert renders an alert component with the title text 1`] = `
"<html><head></head><body><div data-module=\\"dm-alert\\" class=\\"dm-alert dm-alert--success\\">
    <h2 class=\\"dm-alert__title\\">
        A successful action has been completed.
    </h2>
        <div class=\\"dm-alert__body\\">
            This is the text below the title.
        </div>
</div>
</body></html>"
`;
